app/controllers/Users.scala

```
@@ -121,24 +122,28 @@ object Users extends Controller with APIJsonFormats {
     }
   }
 
-  def checkEmail(email: String) = LoggingAction{
+  def checkEmail(email: User.Email) = LoggingAction{
       Action.async { request =>
         User.findByEmail(email).map {
-          case User(Some(`email`), _, _, _, _, _, _, _) :: Nil =>
-            Ok(Json.toJson(TopLevel(emails = Some(Left(Email(email, "registered"))))))
+            // match a list with a single user
+          case user :: Nil =>
+            Ok(Json.toJson(Map("emails" -> Email(email, "registered"))))
+            // match no result or a list bigger than one.
```

You can throw an exception like what you have done in "checkUsername".